### PR TITLE
Add endpoints for widget content generation

### PR DESCRIPTION
**Pull request by d-m-u**
Add ability to trigger generation of widget content from the API. 

~~@miq-bot add_label enhancement, hammer/yes, ivanchuk/yes~~ 

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1623607

@gmcculloug please review, it's your ~~party~~ funeral 
@abellotti please also review, it's less of but still kinda your party

### CODE REVIEW

File: app/controllers/api/widgets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    private
+
+    def ensure_resource_exists(type, id)
+      raise NotFoundError, "#{type} with id:#{id} not found" unless collection_class(type).exists?(id)
+    end
+
+    def generate_widget_content(widget)
+      desc = "#{widget_ident(widget)} content generation"
+      task_id = queue_object_action(widget, desc, :method_name => "generate_content", :role => "")
```
Comment (d-m-u) [inline app/controllers/api/widgets_controller.rb]: Fine, yeah, I have no idea what the role is supposed to be.

**Comment by d-m-u** [comment]
@miq-bot remove_label ivanchuk/yes, hammer/yes
😭

**Comment by gmcculloug** [comment]
@gtanzillo @abellotti Hoping you guys can handle this one as I have little to do with widgets and less to do with the API.  My only contribution to this work is having done something similar in Automate.

@dmetzger57 The BZ is marked as 5.10, but as an enhancement request (with medium priority) I'm guessing maybe we want it in Ivanchuk or just upstream.  Please review.

**Comment by d-m-u** [comment]
@bdunne can you please look again? this is needed for Tuesday's build. thanks.

**Comment by simaishi** [comment]
@d-m-u `ivanchuk/yes` and `hammer/yes` can be put back?

**Comment by d-m-u** [comment]
Ugh I hate it. 
Yes, sorry. 
@miq-bot add_label ivanchuk/yes, hammer/yes

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit d442245656c13931df7fa3ad8041daf4bd5d8993
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Thu Aug 29 09:58:59 2019 -0400

    Merge pull request #660 from d-m-u/adding_widget_endpoint
    
    Add endpoints for widget content generation
    
    (cherry picked from commit 40f597d1aa35693cd2b258b284989aa8f833f01b)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1753682
```

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 0b4c6c7f3fb69e2ade636aba9c477187de06ee43
Author: Gregg Tanzillo <gtanzill@redhat.com>
Date:   Thu Aug 29 09:58:59 2019 -0400

    Merge pull request #660 from d-m-u/adding_widget_endpoint
    
    Add endpoints for widget content generation
    
    (cherry picked from commit 40f597d1aa35693cd2b258b284989aa8f833f01b)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1767837
```

**Pull request merged**